src/modules/builtin/cd.rs

```diff
@@ -0,0 +1,43 @@
+use heraclitus_compiler::prelude::*;
+use crate::modules::expression::expr::Expr;
+use crate::modules::condition::failed::Failed;
+use crate::docs::module::DocumentationModule;
+use crate::translate::module::TranslateModule;
+use crate::utils::{ParserMetadata, TranslateMetadata};
+
+#[derive(Debug, Clone)]
+pub struct Cd {
+    value: Expr,
+    failed: Failed
+}
+
+impl SyntaxModule<ParserMetadata> for Cd {
+    syntax_name!("ChangeDirectory");
+
+    fn new() -> Self {
+        Cd {
+            value: Expr::new(),
+            failed: Failed::new()
+        }
+    }
+
+    fn parse(&mut self, meta: &mut ParserMetadata) -> SyntaxResult {
+        token(meta, "cd")?;
+        syntax(meta, &mut self.value)?;
+        syntax(meta, &mut self.failed)?;
+        Ok(())
+    }
+}
+
+impl TranslateModule for Cd {
+    fn translate(&self, meta: &mut TranslateMetadata) -> String {
+        let value = self.value.translate(meta);
+        format!("cd {} || exit", value)
```

Comment:
You should translate the `self.failed` here as well

Reply:
Shellcheck doesn't know that this will always be handled if the code is written in Amber. We could use a directive for it to ignore this:

![image](https://github.com/user-attachments/assets/49962393-10c1-4a9d-8042-8b6577052d32)

Or we could make it actually `exit` if it fails. I think that most people don't want to handle `cd` as it's so atomic command that we almost always expect it to work. I think that we can remove the failed for this command and just add `cd` that will just exit when failed